src/vs/workbench/contrib/chat/common/chatService.ts

```diff
@@ -106,6 +106,21 @@ export interface IChatProgressMessage {
 	kind: 'progressMessage';
 }
 
+export interface IChatTask extends IChatTaskDto {
+	task?: () => Promise<string | void>;
```

Comment:
Can these be non-optional? I don't see where they are undefined

Reply:
Thanks, fixing both in https://github.com/microsoft/vscode/pull/212372